dbt_semantic_interfaces/implementations/metric.py

```diff
@@ -222,6 +203,27 @@ def _implements_protocol(self) -> Metric:  # noqa: D
     config: Optional[PydanticSemanticLayerElementConfig]
     time_granularity: Optional[str] = None
 
+    @classmethod
+    def parse_obj(cls, input: Any) -> PydanticMetric:
+        """Adds custom parsing to the default method."""
+        data = deepcopy(input)
+
+        # Ensure grain_to_date is lowercased
```

Comment:
Could we lowercase window here too?

Reply:
We could but I think the way it's implemented now is less hacky! This one feels less optimal because we have to go through these dict keys that are not typed. But I couldn't use the same strategy as we did for window because grain is just a string, not an object.